set.mm

```diff
@@ -660788,6 +660948,270 @@ D Fn ( I ... ( M - 1 ) ) /\
     $}
   $}
 
+  ${
+    $d I h $.  $d F b x $.  $d S a b x $.  $d I a b x $.  $d R b x $.
+    $d ph a b x $.  $d W b $.  $d B b $.  $d U b h $.  $d S h $.
+    evlsevl.q $e |- Q = ( ( I evalSub S ) ` R ) $.
+    evlsevl.o $e |- O = ( I eval S ) $.
+    evlsevl.w $e |- W = ( I mPoly U ) $.
+    evlsevl.u $e |- U = ( S |`s R ) $.
+    evlsevl.b $e |- B = ( Base ` W ) $.
+    evlsevl.i $e |- ( ph -> I e. V ) $.
+    evlsevl.s $e |- ( ph -> S e. CRing ) $.
+    evlsevl.r $e |- ( ph -> R e. ( SubRing ` S ) ) $.
+    evlsevl.f $e |- ( ph -> F e. B ) $.
+    $( Evaluation in a subring is the same as evaluation in the ring itself.
+       (Contributed by SN, 9-Feb-2025.) $)
+    evlsevl $p |- ( ph -> ( Q ` F ) = ( O ` F ) ) $=
```

Comment:
Why is it necessary that `S` is a commutative ring?

Reply:
The evaluation functions are defined as the **unique** ring homomorphism that maps polynomials (represented as `(variables --> exponents) --> coefficients`) to functions of the form `(variables --> assignments) --> evaluation` (evlsval and evlsrhm).

If the base ring is not commutative, then the current representation of polynomials as `(variables --> exponents) --> coefficients` is ambiguous. `xy` and `yx` are represented with the same "bag" (`variables --> exponents`), `{x,y} --> {1}`. Therefore, evaluation is ambiguous, and evlsval3, evlsvval, and evlsevl need commutativity.